src/ProxyManager/ProxyGenerator/LazyLoadingGhostGenerator.php

```diff
@@ -60,13 +60,13 @@ class LazyLoadingGhostGenerator implements ProxyGeneratorInterface
     /**
      * {@inheritDoc}
      */
-    public function generate(ReflectionClass $originalClass, ClassGenerator $classGenerator)
+    public function generate(ReflectionClass $originalClass, ClassGenerator $classGenerator, array $properties = [])
     {
         CanProxyAssertion::assertClassCanBeProxied($originalClass);
 
         $interfaces          = [GhostObjectInterface::class];
-        $publicProperties    = new PublicPropertiesMap($originalClass);
-        $privateProperties   = new PrivatePropertiesMap($originalClass);
+        $publicProperties    = new PublicPropertiesMap($originalClass, $properties);
+        $privateProperties   = new PrivatePropertiesMap($originalClass, $properties);
         $protectedProperties = new ProtectedPropertiesMap($originalClass);
```

Comment:
@Ocramius not because, I'm trying to make public and private pass first. But don't happens :'(